src/libp2p/peer_manager.rs

```diff
@@ -212,15 +212,16 @@ impl PeerManager {
 
     /// Removes a peer from the set and returns true if the value was present
     /// previously
-    pub fn mark_peer_bad(&self, peer_id: PeerId) -> bool {
+    pub fn mark_peer_bad(&self, peer_id: PeerId, reason: impl Into<String>) -> bool {
```

Comment:
An idea we can revisit (or not) later - would it make sense to add a label to the bad peers metric so that we see the reason they were banned? This would help with potential debugging when there are a lot of banned peers.

Reply:
That's a good point. I can implement this in a separate PR